packages/core/src/api/webhook.ts

```diff
@@ -0,0 +1,214 @@
+	 * @param data - The data to use when creating the webhook
+	 * @param reason - The reason for creating the webhook
+	 */
+	public async create(channelId: Snowflake, data: RESTPostAPIChannelWebhookJSONBody, reason?: string) {
+		return this.rest.post(Routes.channelWebhooks(channelId), {
+			reason,
+			body: data,
+		}) as Promise<RESTPostAPIWebhookWithTokenResult>;
+	}
+
+	/**
+	 * Edits a webhook
+	 *
+	 * @param id - The id of the webhook to edit
+	 * @param webhook - The new webhook data
+	 * @param options - The options to use when editing the webhook
+	 */
+	public async edit(
+		id: Snowflake,
+		webhook: RESTPatchAPIWebhookJSONBody,
+		{ token, reason }: { reason?: string; token?: string } = {},
+	) {
+		return this.rest.patch(Routes.webhook(id, token), { reason, body: webhook }) as Promise<RESTPatchAPIWebhookResult>;
+	}
+
+	/**
+	 * Deletes a webhook
+	 *
+	 * @param id - The id of the webhook to delete
+	 * @param options - The options to use when deleting the webhook
+	 */
+	public async delete(id: Snowflake, { token, reason }: { reason?: string; token?: string } = {}) {
+		await this.rest.delete(Routes.webhook(id, token), { reason });
+	}
+
+	/**
+	 * Executes a webhook and returns the created message
+	 *
+	 * @param id - The id of the webhook
+	 * @param token - The token of the webhook
+	 * @param data - The data to use when executing the webhook
+	 */
+	public async execute(
+		id: Snowflake,
+		token: string,
+		data: RESTPostAPIWebhookWithTokenJSONBody & RESTPostAPIWebhookWithTokenQuery & { files?: RawFile[]; wait: true },
+	): Promise<RESTPostAPIWebhookWithTokenWaitResult>;
+
+	/**
+	 * Executes a webhook
+	 *
+	 * @param id - The id of the webhook
+	 * @param token - The token of the webhook
+	 * @param data - The data to use when executing the webhook
+	 */
+	public async execute(
+		id: Snowflake,
+		token: string,
+		data: RESTPostAPIWebhookWithTokenJSONBody & RESTPostAPIWebhookWithTokenQuery & { files?: RawFile[]; wait?: false },
+	): Promise<void>;
+
+	/**
+	 * Executes a webhook
+	 *
+	 * @param id - The id of the webhook
+	 * @param token - The token of the webhook
+	 * @param data - The data to use when executing the webhook
+	 */
+	public async execute(
+		id: Snowflake,
+		token: string,
+		{
+			wait,
+			thread_id,
+			files,
+			...body
+		}: RESTPostAPIWebhookWithTokenJSONBody & RESTPostAPIWebhookWithTokenQuery & { files?: RawFile[] },
+	) {
+		return this.rest.post(Routes.webhook(id, token), {
+			query: makeURLSearchParams({ wait, thread_id }),
+			files,
+			body,
+			auth: false,
+			// eslint-disable-next-line @typescript-eslint/no-invalid-void-type
+		}) as Promise<RESTPostAPIWebhookWithTokenWaitResult | void>;
+	}
+
+	/**
+	 * Executes a slack webhook
+	 *
+	 * @param id - The id of the webhook
+	 * @param token - The token of the webhook
+	 * @param options - The options to use when executing the webhook
+	 */
+	public async executeSlack(
+		id: Snowflake,
+		token: string,
+		body: unknown,
```

Comment:
`unknown`...? Is there no type for this?

Reply:
There is, but it's defined by Slack, not discord so it's shape is not provided, as we don't track changes to it's API.